README.mkd

```diff
@@ -242,6 +242,63 @@ As of November 2021 the additional tail-call entry points are only
 implemented in compiler-rt, and calls will only be generated by LLVM
 when the option `-mllvm -save-restore-tailcall` is specified.
 
+## Conventions for vendor extension
+
+Support for custom instruction set extensions are an important part of RISC-V,
+with large encoding spaces reserved of vendor extensions.
+
+However, there are no official guidelines on naming the mnemonics. This section
+defines guidelines which vendors are expected to follow if upstreaming support
+for their extensions. Although vendor-provided toolchains are free to make
+different choices, they are strongly urged to align with these guidelines in
+order to ensure there is a straightforward path for upstreaming in the future.
+
+### Vendor extension naming scheme
+
+According to the RISC-V ISA spec, non-standard extensions are named using a single `x`
+followed by an alphabetical name and an optional version number.
+
+To make it easier to identify and prevent naming conflict, vendor extensions
+should start with a vendor name, which could be an abbreviation of the full name.
+
+For example:
+- `XVentanaCondOps` from Ventana
+- `Xsfcflushdlone` from SiFive
+- `Xtheade` from T-head
+
+### Assembly mnemonic
+
+In order to avoid confusion between standard extension and other vendor
+extensions, instruction mnemonics from vendor extensions must have a
+prefix corresponding to the vendor's name.
+
+e.g. `sf.` for SiFive, `vt.` for Ventana. No central registration with RISC-V
+International or elsewhere is required before the prefix is used.
+
+Vendors should also aim to follow the conventions used for naming mnemonics
+in the ratified base ISA and extensions (e.g. the use of 'w', 'd',
+'u', and 's' suffixes).
+
+The vendor suffix requires at least two letters.
```

Comment:
Isn't the prefix the one that should have two letters? E.g. `sf.magicsauce`?

Reply:
@kito-cheng 'suffix' -> 'prefix'.

Reply:
Yeah...that's really epic typo...